snuba/request/validation.py

```diff
@@ -79,9 +79,9 @@ def build_request(
                 }
                 settings_obj: Union[
                     HTTPRequestSettings, SubscriptionRequestSettings
-                ] = settings_class(**settings)
-            elif isinstance(settings_class, type(SubscriptionRequestSettings)):
-                settings_obj = settings_class(
+                ] = cast(Type[HTTPRequestSettings], settings_class)(**settings)
+            elif settings_class == SubscriptionRequestSettings:
+                settings_obj = cast(Type[SubscriptionRequestSettings], settings_class)(
```

Comment:
Can we change settings_class from `Type[RequestSettings]` to  Type[Union[HTTPRequestSettings, SubscriptionRequestSettings]] so we don't have to cast here?